src/main/java/com/mojang/serialization/MapCompressor.java

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+package com.mojang.serialization;
+
+import it.unimi.dsi.fastutil.ints.Int2ObjectArrayMap;
+import it.unimi.dsi.fastutil.ints.Int2ObjectMap;
+import it.unimi.dsi.fastutil.objects.Object2IntArrayMap;
+import it.unimi.dsi.fastutil.objects.Object2IntMap;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public final class MapCompressor<T> {
```

Comment:
I'd like to keep `Compressor` in the name since it's used for compressing maps down to arrays; how about `StringKeyCompressor` or simply `KeyCompressor`?

Reply:
`KeyCompressors` sounds good